File: backend/src/main/java/com/grupo6daw/lcdd_daw/dto/EventDTO.java
package com.grupo6daw.lcdd_daw.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public record EventDTO(
        Long eventId,
        String eventName,
        String eventDescription,
        ImageDTO eventImage,
        String eventTag,
        boolean requiresRegistration,
        String link,
        UserBasicDTO eventCreator,
        List<NewBasicDTO> eventNews,
        LocalDateTime creationDate,
        LocalDate eventDate,
        Integer maxParticipants,
        boolean validated) {
}

File: frontend/app/dtos/EventDTO.ts
import type NewBasicDTO from "./NewBasicDTO";
import type ImageDTO from "./ImageDTO";
import type UserBasicDTO from "./UserBasicDTO";

export default interface EventDTO {
    eventId: number;
    eventName: string;
    eventDescription: string;
    eventImage: ImageDTO;
    eventTag: string;
    requiresRegistration: boolean;
    link: string;
    eventCreator: UserBasicDTO;
    eventNews: NewBasicDTO[];
    creationDate: string;
    eventDate: string;
    maxParticipants: number;
    validated: boolean;
    participantCount?: number;
    participants?: Array<UserBasicDTO>;
}